openml_OS/models/api/v1/Api_run.php

```diff
@@ -101,7 +103,183 @@ function bootstrap($format, $segments, $request_type, $user_id) {
+   *			  "run_tag": {
+   *			    "id": "2"
+   *			  }
+   *			}
+   *		),
+   *	),
+   *	@OA\Response(
+   *		response=412,
+   *		description="Precondition failed. An error code and message are returned.\n470 - In order to add a tag, please upload the entity id (either data_id, flow_id, run_id) and tag (the name of the tag).\n471 - Entity not found. The provided entity_id {data_id, flow_id, run_id} does not correspond to an existing entity.\n472 - Entity already tagged by this tag. The entity {dataset, flow, run} already had this tag.\n473 - Something went wrong inserting the tag. Please contact OpenML Team.\n474 - Internal error tagging the entity. Please contact OpenML Team.\n",
+   *		@OA\JsonContent(
+   *			ref="#/components/schemas/Error",
+   *		),
+   *	),
+   *)
+   */
+  private function run_tag($run_id, $tag) {
+    $this->run_tag_untag($run_id, $tag, false);
+  }
+
+  /**
+   *@OA\Post(
+   *	path="/run/untag",
+   *	tags={"run"},
+   *	summary="Untag a run",
+   *	description="Untags a run.",
+   *	@OA\Parameter(
+   *		name="run_id",
+   *		in="query",
+   *		@OA\Schema(
+   *          type="integer"
+   *        ),
+   *		description="Id of the run.",
+   *		required=true,
+   *	),
+   *	@OA\Parameter(
+   *		name="tag",
+   *		in="query",
+   *		@OA\Schema(
+   *          type="string"
+   *        ),
+   *		description="Tag name",
+   *		required=true,
+   *	),
+   *	@OA\Parameter(
+   *		name="api_key",
+   *		in="query",
+   *		@OA\Schema(
+   *          type="string"
+   *        ),
+   *		description="Api key to authenticate the user",
+   *		required=true,
+   *	),
+   *	@OA\Response(
+   *		response=200,
+   *		description="The id of the untagged run",
+   *		@OA\JsonContent(
+   *			type="object",
+   *			@OA\Property(
+   *				property="run_untag",
+   *				ref="#/components/schemas/inline_response_200_20_run_untag",
+   *			),
+   *			example={
+   *			  "run_untag": {
+   *			    "id": "2"
+   *			  }
+   *			}
+   *		),
+   *	),
+   *	@OA\Response(
+   *		response=412,
+   *		description="Precondition failed. An error code and message are returned.\n475 - Please give entity_id {data_id, flow_id, run_id} and tag. In order to remove a tag, please upload the entity id (either data_id, flow_id, run_id) and tag (the name of the tag).\n476 - Entity {dataset, flow, run} not found. The provided entity_id {data_id, flow_id, run_id} does not correspond to an existing entity.\n477 - Tag not found. The provided tag is not associated with the entity {dataset, flow, run}.\n478 - Tag is not owned by you. The entity {dataset, flow, run} was tagged by another user. Hence you cannot delete it.\n479 - Internal error removing the tag. Please contact OpenML Team.\n",
+   *		@OA\JsonContent(
+   *			ref="#/components/schemas/Error",
+   *		),
+   *	),
+   *)
+   */
+  private function run_untag($run_id, $tag) {
+    $this->run_tag_untag($run_id, $tag, true);
+  }
 
+  private function run_tag_untag($run_id, $tag, $do_untag) {
```

Comment:
ok now I see what you are doing, LGTM. Except that we can get rid of the 'run_tag_untag' function and make 'run_tag' and 'run_untag' call to entity tag untag immediately.

Reply:
Well I would be fine with leaving the tag_untag function out. However, my thoughts behind this mehtod were to encapsulate the string constants and only have them in one place. But maybe it is a better idea to introduce real constants instead of introducing yet another method. What do you think?